handbook/engineering/environments.md

```diff
@@ -0,0 +1,39 @@
+# Environment
+
+This page describes our different cloud environments.
+
+## Google Cloud
+
+We utilize multiple Google Cloud projects and folders to organize our workloads and manage access control for our engineers, as well as limit the scope of roles and service accounts across projects.
+
+### Root Projects
+These projects contain per-project permissions.
+
+- **TelligentSourcegraph**: Data pipeline and storage for pings and Cloud event logging
+- **Universities**: Sourcegraph instances for universities
+- **sourcegraph-interviews**: Shared project for interviews
+
+### Folders
+
+Dynamic or temporary projects are created in the folders listed below.
+
+### Engineers Projects
+Contains projects used by individual engineers. Engineers are expected to remove all their resources once they are done testing. All projects must be prefixed with `$name-` (the name of the owner).
+
+### Sourcegraph Cloud
+Sourcegraph Cloud projects.
+
+- **Sourcegraph**: Services for sourcegraph.com
+- **Sourcegraph Auxiliary**: Testing clusters, deployments and VMs
+- **sourcegraph-code-intel**: Services for Code Intel code execution
```

Comment:
```suggestion
- **sourcegraph-code-intel**: Services for Code Intel code execution that are separated from our production project for extra isolation.
```

@efritz Can you elaborate why we want this extra isolation?

Reply:
Roles in GCP grant permissions across the entire project, so if we create a role that has DB Admin permissions, it has admin to all databases in that project. GCP Recommends splitting into project, I believe this information might be a bit redundant here given the first section outlines:
> as well as limit the scope of roles and service accounts across projects.

Reply:
Here or in the PR? I think @pecigonzalo, @slimsag, and @chayim could answer this question just as well. I think the main driver was wanting extra isolation to prevent root escalation opportunities from having access to the same GCP APIs in the project that runs our cloud cluster. We also thought it would be nice to separate for better cost accounting so we can determine how much this effort is actually affecting our compute budget.

Reply:
I want to specifically document why code intel is a separate project here on this page since it is the only production service that is separate from the Sourcegraph project. Can either of you push a commit to document this?

Reply:
Will do